core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1663,7 +1663,7 @@ public String getDisplayName() {
         return Messages.Hudson_DisplayName();
     }
 
-    public synchronized List<JDK> getJDKs() {
+    public List<JDK> getJDKs() {
         if(jdks==null)
             jdks = new ArrayList<JDK>();
```

Comment:
this should not be exposed for modification as ArrayList is not ThreadSafe (as commented on in 1947) - but this is a completely different issue.
Any existing code that attempted thread safety by locking on `Jenkins.getInstance()` would still work with this modification.